### ISSUE TITLE
如何构建训练集 How to create training dataset

### ISSUE THREAD

**Issue by venshow-w** [zhuang2002/PowerPaint#113]
您好，非常感谢分享你们的工作，我有几个关于训练集构建的问题：
1. 我在论文上看到除了目标区域的mask之外，还会额外随机mask部分区域，请问这个随机mask区域的text怎么做的呢？是在每次迭代中都随机mask还是先预处理生成一批随机mask的数据，然后训练时读取这个数据？
2. 每次读取的mask是只有一个封闭区域吗？或者说是针对单个目标物体吗？如果是多个目标物体应该怎么构建这个数据集？多目标物体的mask和局部文本位置怎么对应呢？
3. 可以不可以分享几例你们的训练集？不用全部的，只需几个例子就可以，或者数据集构建的说明
感谢！
期待您的回复

**Comment by zengyh1900** [comment]
hi @venshow-w 
1. 随机 mask 的数据是想让模型根据图片已知的内容来推断被mask的内容，所以不需要text; 为了训练稳定，会有一定概率用图像的caption； 随机mask可以在线随机生成，也可以提前生成然后训练时候读取，随机生成有现成的方法；
2. 针对单个物体的 object inpainting 是只考虑了 instance mask 也就是单个实例的；多个实例本工作暂不考虑；你也可以先测试看看是否现有模型能否支持一次inpaint多个实例


比如 训练 P_obj 的数据：
![AP_obj_BP_obj_t1 0_a_boy_and_a_girl_standing_in_front_of_a_door](https://github.com/user-attachments/assets/562473eb-ef87-4d84-8004-13094915a7f4)

训练 P_shape
![AP_shape_BP_ctxt_t0 8932504462075447_a_woman_in_a_white_dress_is_standing_next_to_a_wom](https://github.com/user-attachments/assets/4d10aa76-b189-408b-942d-4a7877bba8e0)

训练 outpainting 就是随机四边有mask
![AP_ctxt_BP_ctxt_t1 0_null](https://github.com/user-attachments/assets/e3815d35-cf38-4f2e-90df-133dd185d274)

**Comment by zqlsnr** [comment]
@zengyh1900 嗨，您好！请问您的意思是不管是inpainting还是outpainting，随机mask构造训练数据，训练时需要图像和图像mask对，都不需要用到text了吗？

**Comment by zengyh1900** [comment]
hi @zqlsnr 
1. 我们的目标是这样的。我们认为随机构造mask的时候，更希望模型能根据没有被遮挡的内容来进行补全，所以训练的时候不加文本的。
2. 但是为了稳定的训练和保留一定的能用文本指引的能力，训练的时候这部分是有一定概率用上image caption 的。

**Comment by huangshihong** [comment]
- name: "openimages"
      # for text-guided task prompts
      prob: 0.4 # the probability of presence during training
      dataset_class: "OpenImageBLIPaug_Dataset"
      desc_prefix: false # set description in front of task prompt

      # you may need to change the path to your own path
      image_root: "openmmlab:s3://openmmlab/datasets/detection/OpenImages/OpenImages/train/"
      mask_root: "data/openimagev6/mask/"
      # prompt that describes segmented objects
      anno_root: "data/openimagev6/prompt"
      aethetic_score_threshold: 0.1
      bufsize: null
      resolution: 512
想问一下openimage中数据集的anno是怎么构建的，        promptA、 promptB 和 prompt 分别指的是什么内容

**Comment by yeluoo** [comment]
> hi @venshow-w
> 
> 1. 随机 mask 的数据是想让模型根据图片已知的内容来推断被mask的内容，所以不需要text; 为了训练稳定，会有一定概率用图像的caption； 随机mask可以在线随机生成，也可以提前生成然后训练时候读取，随机生成有现成的方法；
> 2. 针对单个物体的 object inpainting 是只考虑了 instance mask 也就是单个实例的；多个实例本工作暂不考虑；你也可以先测试看看是否现有模型能否支持一次inpaint多个实例
> 
> 比如 训练 P_obj 的数据： ![AP_obj_BP_obj_t1 0_a_boy_and_a_girl_standing_in_front_of_a_door](https://private-user-images.githubusercontent.com/7473216/378254299-562473eb-ef87-4d84-8004-13094915a7f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5MTA2NjEsIm5iZiI6MTczMTkxMDM2MSwicGF0aCI6Ii83NDczMjE2LzM3ODI1NDI5OS01NjI0NzNlYi1lZjg3LTRkODQtODAwNC0xMzA5NDkxNWE3ZjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMThUMDYxMjQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2VhOTg1YjIwNzViODM0YzEzZWVmNGQxYTc2N2M1YjRkYTIwZmI4ODJiNTczYjU2ZWNhNGNjZGQ4ODNkMjg5OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6l-q-7kuuTYgii7uvRXRYisqargrslutTN0sC-40JXo)
> 
> 训练 P_shape ![AP_shape_BP_ctxt_t0 8932504462075447_a_woman_in_a_white_dress_is_standing_next_to_a_wom](https://private-user-images.githubusercontent.com/7473216/378254504-4d10aa76-b189-408b-942d-4a7877bba8e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5MTA2NjEsIm5iZiI6MTczMTkxMDM2MSwicGF0aCI6Ii83NDczMjE2LzM3ODI1NDUwNC00ZDEwYWE3Ni1iMTg5LTQwOGItOTQyZC00YTc4NzdiYmE4ZTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMThUMDYxMjQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDM0ZjZhY2FjNTY4MGZiMWQyZjY2YjM1MzY3MjkwM2VhZDI1NDZmNTI0Yzc3MGMwNGY2N2Q1YTkzZDRiYzY2NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.2w4hRtPsM9A60NXn-Lf4HDGWTsraxIGVc1ci29EZ8eo)
> 
> 训练 outpainting 就是随机四边有mask ![AP_ctxt_BP_ctxt_t1 0_null](https://private-user-images.githubusercontent.com/7473216/378254882-e3815d35-cf38-4f2e-90df-133dd185d274.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5MTA2NjEsIm5iZiI6MTczMTkxMDM2MSwicGF0aCI6Ii83NDczMjE2LzM3ODI1NDg4Mi1lMzgxNWQzNS1jZjM4LTRmMmUtOTBkZi0xMzNkZDE4NWQyNzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMThUMDYxMjQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Nzg3NGJkZGFkMjg4Y2JkZjU1ZWNiMjdiYmE4NWY2ODA1M2Q4M2NmZWMzOWJiNGRiOTcwODk0MzEzMzViOTcyZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.x6pVOfL4bhIGfxYwFV-giklzQJag6xmpziqLp0HcnS0)

哈喽，想问一下你的object remove的数据长什么样子

**Comment by yanxirao** [comment]
@venshow-w @yeluoo @zengyh1900 大佬我建了一个PPT finetune交流群，大佬能加群，相互交流一下，有些数据处理方式，观看代码，咱们也猜不出来，大家可以一起分享经验，少走一些弯路。
